src/routes/customerCompleted.tsx

```diff
@@ -1,321 +1,321 @@
-    const auth = getAuth();
-    onAuthStateChanged(auth, (currentUser) => {
-      if (currentUser) {
-        console.log("Logged-in user:", currentUser);
-        setUser({
-          email: currentUser.email || "",
-          uid: currentUser.uid,
-          username: currentUser.displayName || "",
-        });
-      } else {
-        setUser(null);
-      }
-    });
-  }, []);
-
-  // Fetch completed packages for the logged-in user
-  useEffect(() => {
-    const fetchCompletedPackages = async () => {
-      if (!user) return; // Wait for user to be set
-
-      try {
-        console.log("Fetching completed packages for user:", user);
-
-        // Step 1: Query the `delivery` collection to get all packages
-        const deliverySnapshot = await getDocs(collection(db, "delivery"));
-
-        if (deliverySnapshot.empty) {
-          console.warn("No packages found in the delivery table.");
-          setCompletedPackages([]);
-          return;
-        }
-
-        const packagesForUser: any[] = [];
-
-        // Step 2: For each delivery document, fetch the email from the `users` table
-        for (const deliveryDoc of deliverySnapshot.docs) {
-          const deliveryData = deliveryDoc.data();
-
-          // Filter by status: Ensure the package status is "completed"
-          if (deliveryData.status !== "completed") continue;
-
-          // Get the user ID (document ID in `users` collection)
-          const userId = deliveryData.user;
-
-          if (!userId) continue;
-
-          // Fetch the corresponding user document
-          const userDocRef = doc(db, "users", userId);
-          const userDoc = await getDoc(userDocRef);
-
-          if (userDoc.exists()) {
-            const userEmail = userDoc.data().email;
-
-            // Match the email with the logged-in user's email
-            if (userEmail === user.email) {
-              // Check if the timestamp is a valid Firestore Timestamp
-              const isTimestamp = deliveryData.timestamp instanceof Timestamp;
-
-              const formattedTimestamp = isTimestamp
-                ? new Intl.DateTimeFormat("en-US", {
-                    dateStyle: "long",
-                    timeStyle: "long",
-                  }).format(deliveryData.timestamp.toDate())
-                : "N/A";
-
-              packagesForUser.push({
-                id: deliveryDoc.id,
-                ...deliveryData,
-                timestampFormatted: formattedTimestamp,
-              });
-            }
-          }
-        }
-
-        console.log(
-          "Fetched completed packages for the logged-in user:",
-          packagesForUser
-        );
-        setCompletedPackages(packagesForUser);
-      } catch (error) {
-        console.error("Error fetching completed packages:", error);
-      }
-    };
-
-    fetchCompletedPackages();
-  }, [user]);
-
-  if (!user) {
-    return (
-      <div className="loading-screen">
-        <p>Loading...</p>
-      </div>
-    );
-  }
```

Comment:
That might be better placed in the root route, where you created the layout, take a look into the auth example in class.